addons/hr_holidays/tests/test_allocations.py

```diff
@@ -177,3 +179,40 @@ def test_disallowed_change_allocation_with_overlapping_allocations(self):
             allocation_one.write({'number_of_days_display': 2, 'number_of_days': 2})
 
         allocation_one.write({'number_of_days_display': 3, 'number_of_days': 3})
+
+    @users('admin')
+    @freeze_time('2024-03-25')
+    def test_allocation_dropdown_after_period(self):
+        """
+        Test when having two allocations of the same type with different
+        time range and submitting a request will the allocations be
+        shown correctly in the dropdown menu or not
+        :return:
+        """
+        allocation = self.env['hr.leave.allocation'].sudo().create({
+            'name': 'Alloc',
+            'employee_id': self.employee.id,
+            'holiday_status_id': self.env.ref('hr_holidays.holiday_status_comp').id,
```

Comment:
Define the ref in a variable above and call it everywhere you need it to avoid multiplying queries